openapi/listings/listings.yaml

```diff
@@ -447,3 +601,297 @@ components:
+            enum:
+              - general
+              - delivery
+              - kitchen
+              - pickup
+              - driveThrough
+              - takeout
+          regularHours:
+            type: array
+            description: |-
+              A collection of times that this location is open for business. Each period represents a range of hours when the location is open during the week.
+
+              You may specify multiple entries for the same date if you are open for multiple periods (Example lunch and supper).
+            items:
+              type: object
+              properties:
+                openDay:
+                  allOf:
+                    - type: string
+                      x-stoplight:
+                        id: 3u07tkjf0bxdg
+                      description: Indicates the day of the week that this period starts on.
+                    - $ref: '#/components/schemas/dayOfTheWeek'
+                closeDay:
+                  allOf:
+                    - type: string
+                      x-stoplight:
+                        id: svnvxunjr767f
+                      description: Indicates the day of the week that this period ends on. It should be the same as `openDay` except when the period goes over midnight. The total period may not be more than 24 hours
+                    - $ref: '#/components/schemas/dayOfTheWeek'
+                openTime:
+                  $ref: '#/components/schemas/timeOfDay'
+                closeTime:
+                  $ref: '#/components/schemas/timeOfDay'
+              required:
+                - openDay
+                - openTime
+                - closeTime
+          specialHours:
+            type: array
+            items:
+              type: object
+              properties:
+                status:
+                  type: string
+                  description: |
+                    Describes the different modes that apply to special hour periods.
+
+                    Currently only `open` and `closed` are supported in the Vendasta platform. We intend to add others in the future.
+                  enum:
+                    - open
+                    - closed
+                startDate:
+                  type: string
+                  format: date
+                  description: The calendar date this special hour period starts on.
+                endDate:
+                  type: string
+                  format: date
+                  description: 'The calendar date this special hour period ends on. If `endDate` field is not set, default to the date specified in `startDate`. If set, this field must be equal to or at most 1 day after `startDate`.'
+                openTime:
+                  $ref: '#/components/schemas/timeOfDay'
+                closeTime:
+                  $ref: '#/components/schemas/timeOfDay'
+              required:
+                - status
+                - startDate
+                - endDate
```

Comment:
The description for endDate says that it defaults to the startDate, which implies that the field should be optional.